### PR TITLE
Add animation on disclosure icon

### LINKED ISSUE #2901: [Enhancement] Animation on disclosure icon, on kirby item
We are using the new disclosure icon on menu elements  to indicate state for a fold state. Previously we had an animation that is no longer possible if we use the disclosure option.

## Describe the enhancement
A way to animate discloure icons

## Describe the solution you'd like
A prebuilt rotate option or other way to pass an animation config to the disclosure icon.

## Have you considered any alternatives?
Not using disclure works today, but requires extra styling to match the items that can just disclosure, as is.

## Are there any additional context?
Previous solution
https://user-images.githubusercontent.com/110398431/221561601-235f678b-5b57-4ee5-bfff-05917d73bf6a.mov

Simply using disclosure
https://user-images.githubusercontent.com/110398431/221561614-d9cb1caf-7375-496f-af40-b9d487ff71b0.mov

<hr />

## Checklist:

The following tasks should be carried out in sequence in order to follow [the process of contributing](https://github.com/kirbydesign/designsystem/blob/main/.github/CONTRIBUTING.md/#the-process-of-contributing) correctly.

### Refinement

- [x] Request that the issue is [UX refined](https://github.com/kirbydesign/designsystem/blob/main/.github/CONTRIBUTING.md/#ux-refinement); do not proceed until this is done.
- [x] Request that the issue is [tech refined](https://github.com/kirbydesign/designsystem/blob/main/.github/CONTRIBUTING.md/#tech-refinement); do not proceed until this is done.

### Implementation 
The contributor who wants to implement this issue should: 

- [x] Make sure you have read: "[Before you get coding](https://github.com/kirbydesign/designsystem/blob/main/.github/CONTRIBUTING.md/#before-you-get-coding)".
- [x] Signal to others you are working on the issue by assigning yourself.
- [x] Create a branch from the [develop branch](https://github.com/kirbydesign/designsystem/tree/develop) following our [branch naming convention](https://github.com/kirbydesign/designsystem/wiki/The-Good%3A-Branch). 
- [x] Publish a WIP implementation to Github as a draft PR and ask for feedback. 
- [x] Make sure you have implemented tests following the guidelines in: "[The good: Test](https://github.com/kirbydesign/designsystem/wiki/The-Good%3A-Test)".
- [ ] Update the [cookbook](https://cookbook.kirby.design) with examples and showcases.

### Review
Once the issue has been implemented and is ready for review:

- [ ] Do a [self-review](https://github.com/kirbydesign/designsystem/wiki/The-Good%3A-Self-review). 
- [ ] Create a pull-request. If you created a draft PR during implementation you can just mark that as "ready for review".

### PR DESCRIPTION
**Pull request by mark-drastrup**
## Which issue does this PR close?

This PR closes #2901

## What is the new behavior?
The `arrow-up` and `arrow-down` disclosure icons on `kirby-icon` now have a 180deg rotation animation, when `rotateIcon` is set to `true.`

## Does this PR introduce a breaking change?

- [ ] Yes
- [x] No

## Checklist:

The following tasks should be carried out in sequence in order to follow [the process of contributing](https://github.com/kirbydesign/designsystem/blob/main/.github/CONTRIBUTING.md/#the-process-of-contributing) correctly.

### Reminders
- [x] Make sure you have implemented tests following the guidelines in: "[The good: Test](https://github.com/kirbydesign/designsystem/wiki/The-Good%3A-Test)".
- [x] Make sure you have updated the cookbook with examples and showcases (for bug fixes, enhancements & new components).

### Review  
- [x] Determine if your changes are a fix, feature or breaking-change, and add the matching label to your PR. If it is tooling, dependency updates or similar, add ignore-for-release.
- [x] Do a [self-review](https://github.com/kirbydesign/designsystem/wiki/The-Good%3A-Self-review).
- [x] Request that the changes are code-reviewed 
- [x] Request that the changes are [UX reviewed](https://github.com/kirbydesign/designsystem/blob/main/.github/CONTRIBUTING.md/#ux-review) (only necessary if your PR introduces visual changes)

When the pull request has been approved it will be merged to `develop` by Team Kirby.

### CODE REVIEW

File: libs/designsystem/item/src/item.component.html [hunk 1]
```diff
@@ -14,5 +14,10 @@
     <kirby-icon name="reorder"></kirby-icon>
   </ion-reorder>
 
-  <kirby-icon *ngIf="disclosure" class="disclosure" slot="end" name="{{ disclosure }}"></kirby-icon>
+  <span class="disclosure" slot="end" *ngIf="disclosure">
```
Comment (mark-drastrup) [inline libs/designsystem/item/src/item.component.html:17]: This addition `span` is needed to make the animation look correct. The element with `slot="end"` needs to have a left padding of 4px to be positioned correctly relative to the content to the left. If we do the rotation on the same element that has the left padding, then the axis of the rotation seems off.

File: libs/designsystem/item/src/item.component.spec.ts [hunk 2]
```diff
@@ -92,6 +92,52 @@ describe('ItemComponent', () => {
...
+        spectator.setInput('rotateIcon', true);
+      });
+
+      it('should rotate the icon 180deg if the disclosure icon is "arrow-down', () => {
+        spectator.setInput('disclosure', 'arrow-down');
+
+        const icon = spectator.query('kirby-icon');
+
+        expect(icon).toHaveComputedStyle({
+          transform: 'matrix(-1, 0, 0, -1, 0, 0)',
```
Comment (mark-drastrup) [inline libs/designsystem/item/src/item.component.spec.ts:107]: In the `computed` tab in the browser `rotate(180deg)` is converted to `matrix(-1, 0, 0, -1, 0, 0)`. Is there a better (and less confusing) way to test this?

File: apps/cookbook/src/app/examples/item-example/examples/disclosure-animation.ts [hunk 3]
```diff
@@ -0,0 +1,21 @@
+import { Component } from '@angular/core';
+
+const config = {
+  selector: 'cookbook-item-example-disclosure-animation',
+  template: `<kirby-item selectable="true" [disclosure]="'arrow-down'" [rotateIcon]="rotateIcon" (click)="rotateIcon = !rotateIcon">
```
Comment (jakobe) [inline apps/cookbook/src/app/examples/item-example/examples/disclosure-animation.ts]: Consider simplifying as it's just a string:
```suggestion
  template: `<kirby-item selectable="true" disclosure="arrow-down" [rotateIcon]="rotateIcon" (click)="rotateIcon = !rotateIcon">
```

File: apps/cookbook/src/app/showcase/item-showcase/item-showcase.component.html [hunk 4]
```diff
@@ -270,6 +270,29 @@ <h2 #disclosure>Disclosure</h2>
   </em>
 </p>
 
+<h4>Rotation animation</h4>
+
+<p>
+  It is possible to enable an animation that rotates the discloure icon 180 degrees. Pass a boolean
```
Comment (jakobe) [inline apps/cookbook/src/app/showcase/item-showcase/item-showcase.component.html]: Consider clarifying that the boolean should be a flag, that the consuming code controls.
The wording could be interpreted as if the boolean is to turn the animation on/off once and for all, i.e. `[rotateIcon]=true`

File: apps/cookbook/src/app/examples/item-example/examples/disclosure-animation.ts [hunk 5]
```diff
@@ -0,0 +1,21 @@
...
+    <kirby-avatar overlay="true" slot="start">
+        <kirby-icon name="moneybag"></kirby-icon>
+    </kirby-avatar>
+  <h3>Title</h3>
+  </kirby-item>`,
```
Comment (jakobe) [inline apps/cookbook/src/app/examples/item-example/examples/disclosure-animation.ts]: Consider adding additional item(s) that gets expanded/collapsed on click to showcase a real-world scenario - potentially wrapped in a card or using our list...

File: apps/cookbook/src/app/showcase/item-showcase/item-showcase.component.html [hunk 6]
```diff
@@ -270,6 +270,29 @@ <h2 #disclosure>Disclosure</h2>
   </em>
 </p>
 
+<h4>Rotation animation</h4>
```
Comment (jakobe) [inline apps/cookbook/src/app/showcase/item-showcase/item-showcase.component.html:273]: Consider adding margin-top to `h4` in the showcase scss - currently it has the same visual space before and after which makes it hard to see what belongs together:
image

File: libs/designsystem/item/src/item.component.html (hunk 1 above)
Comment (jakobe) [reply libs/designsystem/item/src/item.component.html:17]: Nice catch 👌🏻

File: libs/designsystem/item/src/item.component.scss [hunk 7]
```diff
@@ -94,6 +94,15 @@
 .disclosure {
   padding-left: #{utils.size('xxxs')};
   color: #{utils.get-color('semi-dark')};
+  display: inline-flex;
```
Comment (jakobe) [inline libs/designsystem/item/src/item.component.scss:102]: Why is this needed? 🤔

File: libs/designsystem/item/src/item.component.spec.ts [hunk 8]
```diff
@@ -6,10 +6,13 @@ import { DesignTokenHelper } from '@kirbydesign/designsystem/helpers';
 
 import { IconModule } from '@kirbydesign/designsystem/icon';
 
+import { TestHelper } from 'testing/src';
```
Comment (jakobe) [inline libs/designsystem/item/src/item.component.spec.ts]: Probably doesn't matter but everywhere else we use the absolute import:
```suggestion
import { TestHelper } from '@kirbydesign/designsystem/testing';
```

File: libs/designsystem/item/src/item.component.spec.ts [hunk 9]
```diff
@@ -92,6 +95,54 @@ describe('ItemComponent', () => {
...
+
+        expect(icon).toHaveComputedStyle({
+          transform: rotationMatrix,
+        });
+      });
+
+      it('should NOT rotate the icon 180deg if the disclosure icon is "arrow-more', () => {
+        spectator.setInput('disclosure', 'arrow-more');
+
+        const icon = spectator.query('kirby-icon');
```
Comment (jakobe) [inline libs/designsystem/item/src/item.component.spec.ts:141]: I'm guessing you should add a tick here as well, otherwise you could have a false positive on the `transform: none`:
```suggestion
        const icon = spectator.query('kirby-icon');
        await TestHelper.waitForTimeout(); // Wait a tick
```

File: libs/designsystem/item/src/item.component.spec.ts [hunk 10]
```diff
@@ -92,6 +95,54 @@ describe('ItemComponent', () => {
...
+
+        expect(icon).toHaveComputedStyle({
+          transform: 'none',
+        });
+      });
+
+      it('should NOT rotate the icon 180deg if the disclosure icon is "link', () => {
+        spectator.setInput('disclosure', 'link');
+
+        const icon = spectator.query('kirby-icon');
```
Comment (jakobe) [inline libs/designsystem/item/src/item.component.spec.ts:152]: I'm guessing you should add a tick here as well, otherwise you could have a false positive on the `transform: none`:
```suggestion
        const icon = spectator.query('kirby-icon');
        await TestHelper.waitForTimeout(); // Wait a tick
```

**Review by jakobe — CHANGES_REQUESTED** [review]
Good work! just the usual nitty gritties ;)

File: apps/cookbook/src/app/showcase/item-showcase/item-showcase.component.html [hunk 11]
```diff
@@ -270,6 +270,35 @@ <h2 #disclosure>Disclosure</h2>
...
+
+<p>
+  It is possible to trigger an animation that rotates the disclosure icon by passing a boolean flag
+  to
+  <code>[rotateIcon]</code>
+  .
```
Comment (jakobe) [inline apps/cookbook/src/app/showcase/item-showcase/item-showcase.component.html]: Consider clarifying:
```suggestion
  . When set to <code>true</code> it will rotate the icon 180 degrees and when set to <code>false</code> it will rotate the icon back to it's original position.
```

File: apps/cookbook/src/app/examples/item-example/examples/disclosure-animation.ts [hunk 12]
```diff
@@ -0,0 +1,40 @@
...
+  selector: config.selector,
+  template: config.template,
+  styles: [':host {--kirby-card-padding-top: 0; --kirby-card-padding-bottom: 0;}'],
+  animations: [
+    trigger('isExpanded', [
+      state('true', style({ height: '*', visibility: 'visible' })),
+      state('false', style({ height: '0px', visibility: 'hidden' })),
+      transition('true <=> false', animate('0.2s')),
+    ]),
+  ],
```
Comment (jakobe) [inline apps/cookbook/src/app/examples/item-example/examples/disclosure-animation.ts:45]: Consider adding this as a code snippet to the showcase, otherwise it might a bit black magic how/why the `[@isExpanded]` binding works in the markup / html snippet...

File: apps/cookbook/src/app/examples/item-example/examples/disclosure-animation.ts [hunk 13]
```diff
@@ -0,0 +1,40 @@
...
+  </kirby-card-header>
+
+    <div [@isExpanded]="isExpanded">
+      <p class="kirby-text-large">A card with animated disclosure icon</p>
+      <p> 
+        lorem ipsum dolor sit amet, consectetur adipiscing elit.
+      </p>
+    </div>
+  </kirby-card>
+  `,
```
Comment (jakobe) [inline apps/cookbook/src/app/examples/item-example/examples/disclosure-animation.ts]: Be aware that linebreak and whitespace renders in the code viewer:
```suggestion
  template: `<kirby-card [hasPadding]="true" (click)="isExpanded = !isExpanded">
  <kirby-card-header [hasPadding]="false">
    <kirby-item disclosure="arrow-down" [rotateIcon]="isExpanded">
      <p class="kirby-text-bold">Disclosure animation example</p>
    </kirby-item>
  </kirby-card-header>

  <div [@isExpanded]="isExpanded">
    <p class="kirby-text-large">A card with animated disclosure icon</p>
    <p>Lorem ipsum dolor sit amet, consectetur adipiscing elit.</p>
  </div>
</kirby-card>`,
```

File: apps/cookbook/src/app/showcase/item-showcase/item-showcase.component.html [hunk 14]
```diff
@@ -270,6 +270,29 @@ <h2 #disclosure>Disclosure</h2>
...
+<cookbook-example-viewer [html]="disclosureAnimationExample.template">
+  <cookbook-item-example-disclosure-animation
+    #disclosureAnimationExample
+  ></cookbook-item-example-disclosure-animation>
+</cookbook-example-viewer>
+
+<p class="comment small">
+  <em>
+    <strong>Please note</strong>
+    , the animation of the disclosure icon is only available for:
```
Comment (jakobe) [inline apps/cookbook/src/app/showcase/item-showcase/item-showcase.component.html]: Consider readability:
```suggestion
    <strong>Please note:</strong>
    the animation of the disclosure icon is only available for:
```

**Review by jakobe — CHANGES_REQUESTED** [review]

File: libs/designsystem/item/src/item.component.scss [hunk 15]
```diff
@@ -94,6 +94,15 @@
 .disclosure {
   padding-left: #{utils.size('xxxs')};
   color: #{utils.get-color('semi-dark')};
+  line-height: 0;
```
Comment (jakobe) [inline libs/designsystem/item/src/item.component.scss:97]: Consider changing to:
```suggestion
  display: inline-block;
  line-height: 1;
```

File: apps/cookbook/src/app/examples/item-example/examples/disclosure-animation.ts [hunk 16]
```diff
@@ -0,0 +1,54 @@
...
+    <kirby-item disclosure="arrow-down" [rotateIcon]="isExpanded">
+      <p class="kirby-text-bold">Disclosure animation example</p>
+    </kirby-item>
+  </kirby-card-header>
+
+  <div [@isExpanded]="isExpanded">
+    <p class="kirby-text-large">A card with animated disclosure icon</p>
+    <p> 
+      lorem ipsum dolor sit amet, consectetur adipiscing elit.
+    </p>
```
Comment (jakobe) [inline apps/cookbook/src/app/examples/item-example/examples/disclosure-animation.ts]: Consider semantics and visual levels:
```suggestion
      <h3 class="kirby-text-bold">Disclosure animation example</h3>
    </kirby-item>
  </kirby-card-header>

  <div [@isExpanded]="isExpanded">
    <p class="kirby-text-bold">A card with animated disclosure icon</p>
    <p>Lorem ipsum dolor sit amet, consectetur adipiscing elit.</p>
```

File: apps/cookbook/src/app/showcase/item-showcase/item-showcase.component.html [hunk 17]
```diff
@@ -276,19 +276,26 @@ <h4>Rotation animation</h4>
...
   <cookbook-item-example-disclosure-animation
     #disclosureAnimationExample
   ></cookbook-item-example-disclosure-animation>
 </cookbook-example-viewer>
 
 <p class="comment small">
   <em>
-    <strong>Please note</strong>
-    , the animation of the disclosure icon is only available for:
+    <strong>Please note:</strong>
```
Comment (jakobe) [inline apps/cookbook/src/app/showcase/item-showcase/item-showcase.component.html:297]: Great 👌🏻 
Could you maybe change it also for the note under "Disclosure"? 🙏🏻 
image

File: libs/designsystem/item/src/item.component.spec.ts [hunk 18]
```diff
@@ -92,6 +95,66 @@ describe('ItemComponent', () => {
...
     });
+
+    it('should make the disclosure wrapper element get the same height as its child (kirby-icon)', async () => {
+      spectator.setInput('disclosure', 'arrow-down');
+      spectator.detectChanges();
+
+      const disclosureWrapper = spectator.query<HTMLElement>('.disclosure');
+      const icon = spectator.query<HTMLElement>('kirby-icon');
+
+      expect(getComputedStyle(disclosureWrapper).height).toBe(getComputedStyle(icon).height);
```
Comment (jakobe) [inline libs/designsystem/item/src/item.component.spec.ts]: Consider simplifying and using existing test utils:
```suggestion
      expect(disclosureWrapper).toHaveComputedStyle(
        {
          height: getComputedStyle(icon).height),
        }
      );
```

**Review by jakobe — CHANGES_REQUESTED** [review]

**Review by jakobe — APPROVED** [review]
Hoooooooraaaaaayyyyyyy 🥳 🥳 🥳 🥳 🥳 🥳 🥳 🥳 🥳
